File: Java/Stacks/src/StackTest.java
import org.junit.*;

import static org.junit.Assert.fail;

public class StackTest {

    static Stack<String> stack = new Stack<String>(10);

    @Before
    public void setup(){
        System.out.println("-----------------------SETUP---------------------------------------");
        for(int i = 0; i <= 8; i++){
            stack.push("test" + i);
        }
        System.out.println(stack.toString());
        System.out.println("Initial length: " + stack.length());
        stack.showTop();
        System.out.println("----------------------END SETUP------------------------------------");
    }

    @After
    public void report(){
        System.out.println("-----------------------REPORT---------------------------------------");
        System.out.println(stack.toString());
        System.out.println("Final length: " + stack.length());
        stack.showTop();

        int i = 0;
        while(i < stack.length()){
            try {
                stack.removeTop();
            }catch (NoElementStackedException exception){
                System.out.println("Stack is empty");
            }
        }
        i = 0;
        System.out.println("----------------------END REPORT------------------------------------");
    }

    @Test
    public void ShouldInvertTheOrderOfTheStack(){
        System.out.println("Testing invert() method");
        String sub1 = stack.toString();
        System.out.println(sub1);
        stack.invert();
        String sub2 = stack.toString();
        System.out.println(sub2);
        assert sub1 != sub2;
    }

    @Test
    public void ShouldGetTheItemOnTheTopOfTheStack(){
        System.out.println("Testing getTop() method");
        try {
            assert stack.getTop().equals("test8");
        }catch (NoElementStackedException exception){
            fail();
        }
    }

    @Test
    public void ShouldAddAnItemOnTheTopOfTheStack(){
        System.out.println("Testing push() method");
        stack.push("TestDummy");
        try {
            assert stack.getTop().equals("TestDummy");
        }catch (NoElementStackedException exception){
            fail();
        }
    }

    @Test
    public void ShouldReturnFalseIfStackHasValues(){
        System.out.println("Testing isEmpty() method");

        assert !stack.isEmpty(); //not true
    }
}

File: Kotlin/Stacks/src/StackTest.kt
import Stack.NoElementStackedException
import org.junit.After
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4
import org.junit.Assert.*

@RunWith(JUnit4::class)
class StackTest {

    companion object{
        val stack = Stack<String>(10)
    }

    @Before
    fun setup(){
        println("-----------------------SETUP---------------------------------------")
        for(i: Int in 0..8){
            stack.push("test${i}")
        }
        println(stack.toString())
        println("Initial length: ${stack.length()}")
        stack.showTop()
        println("---------------------END SETUP-------------------------------------")
    }

    @After
    fun report(){
        println("-----------------------REPORT--------------------------------------")
        println(stack.toString())
        println("Final length: ${stack.length()}")
        stack.showTop()

        for(i: Int in 0..stack.length()){
            stack.removeTop()
        }
        println("---------------------END REPORT------------------------------------")
    }

    @Test
    fun `should invert the order of the stack`(){
        println("Testing invert() method")
        val sub1 = stack.toString()
        println(stack.toString())
        stack.invert()
        println(stack.toString())
        val sub2 = stack.toString()
        assertNotEquals(sub1, sub2)
    }

    @Test
    fun `should get the item on the top of the stack`(){
        println("Testing getTop() method")
        try{
            assertEquals("test8", stack.getTop())
        }catch(e: NoElementStackedException){
            fail()
        }
    }

    @Test
    fun `should add an item on the top of the stack`(){
        println("Testing push() method")
        stack.push("TestDummy")
        val res = stack.getTop()
        assertEquals("TestDummy", res)
    }

    @Test
    fun `should return false if stack has values`(){
        println("Testing isEmpty() method")

        assertEquals(false, stack.isEmpty())
    }


}